docs/howto_guides/reproject_vbi_moasic.md

```diff
@@ -0,0 +1,148 @@
+
+from sunpy.net import Fido, attrs as a
+
+import dkist
+import dkist.net
+```
+
+```{code-cell} ipython3
+res = Fido.search(a.dkist.Dataset("AJQWW"), )
+res
+```
+
+Let's download a VBI dataset, which has good seeing:
+
+```{code-cell} ipython3
+asdf_file = Fido.fetch(res, path="~/dkist_data/{dataset_id}")
+asdf_file
+```
+
+This gives us a `TiledDataset` object, which as we have already seen is like an array of `Dataset` objects.
+
+```{code-cell} ipython3
+ds = dkist.load_dataset(asdf_file)
+ds
+```
+
+This mosaic is 3 x 3:
+
+```{code-cell} ipython3
+ds.shape
+```
+
+To start off, let's just download the first frame of each tile.
+
+```{code-cell} ipython3
+first_tiles = [tile[0] for tile in ds.flat]
+```
+
+```{code-cell} ipython3
+:tags: [skip-execution]
+
+for i, tile in enumerate(first_tiles):
+    tile.files.download(wait=i == len(first_tiles) - 1)
+```
+
+We can now make a composite plot of all the tiles.
+
+```{code-cell} ipython3
+fig = plt.figure(figsize=(12, 12))
+
+for i, tile in enumerate(first_tiles):
+    ax = fig.add_subplot(ds.shape[0], ds.shape[1], i+1, projection=tile.wcs)
+    ax.set_title(f"MINDEX1={tile.headers[0]['MINDEX1']}, MINDEX2={tile.headers[0]['MINDEX2']}")
+    ax.imshow(tile.data)
+
+fig.tight_layout()
+```
+
+## Regridding with Reproject
+
+First, we need the most recent version of reproject (released on Friday) for this to work, so let's check:
+
+```{code-cell} ipython3
+import reproject
+```
+
+Now import everything we need for it to run
+
+```{code-cell} ipython3
+from reproject.mosaicking import find_optimal_celestial_wcs, reproject_and_coadd
+from reproject import reproject_interp
+
+from ndcube import NDCube
+```
+
+First, let us crop off the edges of all our tiles to remove some artifacts.
+
+```{code-cell} ipython3
+first_tiles = [d[0, 100:-100, 100:-100] for d in ds.flat]
+```
+
+Next we need to calculate the optimal WCS for the output:
+
+```{code-cell} ipython3
+reference_wcs, shape_out = find_optimal_celestial_wcs(
+    [f.wcs for f in first_tiles],
+    auto_rotate=True,
+    # We drop the output resolution by a factor of 10 to reduce memory
+    # remove this line to run at the native resolution of the input data
+    resolution=0.1*u.arcsec,
+)
+
+# Due to a bug in reproject we need to reverse the direction of the longitude axis
+# https://github.com/astropy/reproject/issues/431
+reference_wcs.wcs.cdelt[0] = -reference_wcs.wcs.cdelt[0]
+```
+
+Now we can do the actual reprojection
+
+```{code-cell} ipython3
+arr, footprint = reproject_and_coadd(
+    first_tiles,
+    reference_wcs,
+    reproject_function=reproject_interp,
+    shape_out=shape_out,
+    roundtrip_coords=False,
+)
+```
+
+Make a new `NDCube` object and plot it.
+
+```{code-cell} ipython3
+plt.figure(figsize=(10,10))
+stitched = NDCube(arr, reference_wcs)
+stitched.plot()
+```
+
+We can see some small artifacting which can be hidden by using the `match_background=True` keyword argument to `reproject_and_coadd`.
```

Comment:
Is it worth explaining more what this argument does?

Reply:
Perhaps we should just remove this line, you don't really see the artifacting in this data.